VERSIONING.md

```diff
@@ -0,0 +1,95 @@
+
+We (mostly) follow the [common KubeBuilder versioning
+guidelines][guidelines], and use the corresponding tooling and PR process
+described there.
+
+For the purposes of the aforementioned guidelines, KubeBuilder counts as
+a "CLI project".
+
+[guidelines]: https://sigs.k8s.io/kubebuilder-release-tools/VERSIONING.md
+
+## Compability
+
+Note that we generally do not support older release branches, except in
+extreme circumstances.
+
+Bear in mind that changes to scaffolding generally constitute breaking
+changes -- see [below](#understanding-the-versions) for more details.
+
+## Releasing
+
+When releasing, you'll need to:
+
+- to update references in [the build directory](build/) to the latest
+  version of the [envtest tools](#tools-releases) **before tagging the
+  release.**
+
+- reset the book branch: see [below](#book-releases)
+
+You may also want to check that the book is generating the marker docs off
+the latest controller-tools release.  That info is stored in
+[docs/book/install-and-build.sh](/docs/book/install-and-build.sh).
+
+## Book Releases
+
+The book's main version (https://book.kubebuilder.io) is published off of
+the [book-v2][book-branch] (a version built off the main branch can be
+found at https://master.book.kubebuilder.io).
+
+Docs changes that aren't specific to a new feature should be
+cherry-picked to the aforementioned branch to get them to be published.
+The cherry-picks will automatically be published to the book once their PR
+merges.
+
+**When you publish a KubeBuilder release**, be sure to also submit a PR
+that merges the main branch into [book-v2][book-branch], so that it
+describes the latest changes in the new release.
+
+[book-branch]: https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases
+
+## Tools Releases
+
+In order to update the [envtest tools][envtest-ref], you'll need to do an
+update to the [tools-releases branch][tools-branch].  Simply submit a PR
+against that branch that changes all references to the current version to
+the desired next version.  Once the PR is merged, Google Cloud Build will
+take care of building and publishing the artifacts.
+
+[envtest-ref]: https://book.kubebuilder.io/reference/artifacts.html
+[tools-branch]: https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases)
+
+## Understanding the versions
+
+|   Name	|   Example	|  Description |
+|---	|---	|---	|
+|  PROJECT version |  `v1`,`v2`,`v3` | As of the introduction of [Extensible CLI and Scaffolding Plugins](https://github.com/kubernetes-sigs/kubebuilder/blob/master/designs/extensible-cli-and-scaffolding-plugins-phase-1.md), PROJECT version represents the layout of PROJECT file itself.  For `v1` and `v2` projects, there's extra meaning -- see below.  |
+|  Release version | `v2.2.0`, `v2.3.0`, `v2.3.1` | Tagged versions of the KubeBuilder project, representing changes to the source code in this repository. See the [releases](https://github.com/kubernetes-sigs/kubebuilder/releases) page. |
```

Comment:
We need to change these content, however, it is not part of the scope of this Pr so. 👍